chart.js.d.ts

```diff
@@ -0,0 +1,605 @@
+      hitRadius?: number
+      hoverRadius?: number
+      hoverBorderWidth?: number
+    }
+    rectangle?: {
+      backgroundColor?: string
+      borderWidth?: number
+      borderColor?: string
+      borderSkipped?: string
+    }
+  }
+
+  interface Element {
+
+  }
+
+  class ChartInstance {
+    destroy(): void;
+    update(duration?: number, lazy?: boolean): void;
+    render(duration?: number, lazy?: boolean): void;
+    stop(): void;
+    resize(): void;
+    clear(): void;
+    toBase64Image(): string;
+    generateLegend(): string;
+    getElementAtEvent(e: Event): Element; // Check required. What does element return type contain? is Event correct here?
+    getElementsAtEvent(e: Event): Array<Element>; // Check required
+    getDatasetAtEvent(e: Event): Array<Element>; // Check required
+  }
+
+  interface LineChartData extends ChartData {
+    datasets?: Array<ChartDataSet & {
+      label?: string
+      fill?: boolean
+      backgroundColor?: string
+      borderColor?: string | Array<string>
+      borderCapStyle?: string
+      borderDash?: Array<number>
+      borderDashOffset?: number
+      borderJoinStyle?: string
+      pointBorderColor?: string | Array<string>
+      pointBackgroundColor?: string | Array<string>
+      pointBorderWidth?: number | Array<number>
+      pointHoverRadius?: number | Array<number>
+      pointHoverBackgroundColor?: string | Array<string>
+      pointHoverBorderColor?: string | Array<string>
+      pointHoverBorderWidth?: number | Array<number>
+      tension?: number
+      yAxisID?: string
+    }>
+  }
+
+  interface LineChartOptions extends ChartOptions {
+    showLines?: boolean
+    stacked?: boolean
+    scales?: {
+      xAxes?: Array<ScaleOptions & {
+        id?: string
+      }>
+      yAxes?: Array<ScaleOptions & {
+        id?: string
+      }>
+    }
+  }
+
+  class LineChartInstance extends ChartInstance {
+    data: LineChartData;
+    options: LineChartOptions;
+  }
+
+  interface BarChartData extends ChartData {
+    datasets?: Array<ChartDataSet & {
+      label?: string
+      backgroundColor?: string | Array<string>
+      borderColor?: string | Array<string>
+      borderWidth?: number | Array<number>
+      hoverBackgroundColor?: string | Array<string>
+      hoverBorderColor?: string | Array<string>
+      yAxisID?: string
+    }>
+  }
+
+  interface BarChartOptions extends ChartOptions {
+    stacked?: boolean
+    scales?: Array<{
+      xAxes?: Array<ScaleOptions & {
+        position?: string
+        id?: string
+        categoryPercentage?: number
+        barPercentage?: number
+        gridLines?: Array<ScaleGridLinesOptions & {
+          offsetGridLines?: boolean
+        }>
+      }>
+      yAxes?: Array<ScaleOptions & {
+        position?: string
+        id?: string
+      }>
+    }>
+  }
+
+  class BarChartInstance extends ChartInstance {
+    data: BarChartInstance;
+    options: BarChartOptions;
+  }
+
+  interface RadarChartData extends ChartData {
```

Comment:
@DaanDeMeyer how do you think something like this looks? This is for the Line Chart.

### Data structure

The following options can be included in a line chart dataset to configure options for that specific dataset.

All point* properties can be specified as an array. If these are set to an array value, the first value applies to the first point, the second value to the second point, and so on.

Property | Type | Usage
--- | --- | ---
data | `Array<Number>` | The data to plot in a line
label | `String` | The label for the dataset which appears in the legend and tooltips
xAxisID | `String` | The ID of the x axis to plot this dataset on
yAxisID | `String` | The ID of the y axis to plot this dataset on
fill | `Boolean` | If true, fill the area under the line
lineTension | `Number` | Bezier curve tension of the line. Set to 0 to draw straightlines. *Note* This was renamed from 'tension' but the old name still works.
backgroundColor | `Color` | The fill color under the line. See [MDN](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/fillStyle) for supported options
borderColor | `Color` | The color of the line.
borderCapStyle | `String` | Cap style of the line. See [MDN](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/lineCap)
borderDash | `Array<Number>` | Length and spacing of dashes. See [MDN](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/setLineDash)
borderDashOffset | `Number` | Offset for line dashes. See [MDN](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/lineDashOffset)
borderJoinStyle | `String` | Line joint style. See [MDN](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/lineJoin)
pointBorderColor | `Color or Array<Color>` | The border color for points.
pointBackgroundColor | `Color or Array<Color>` | The fill color for points
pointBorderWidth | `Number or Array<Number>` | The width of the point border in pixels
pointRadius | `Number or Array<Number>` | The radius of the point shape. If set to 0, nothing is rendered. 
pointHoverRadius | `Number or Array<Number>` | The radius of the point when hovered
pointHitRadius | `Number or Array<Number>` | The pixel size of the non-displayed point that reacts to mouse events
pointHoverBackgroundColor | `Color or Array<Color>` | Point background color when hovered
pointHoverBorderColor | `Color or Array<Color>` | Point border color when hovered
pointHoverBorderWidth | `Number or Array<Number>` | Border width of point when hovered